src/OpenSage.Game/Data/Wav/WavFile.cs

```diff
@@ -65,51 +135,118 @@ public sealed class WavFile
     {
         private RiffHeader _header;
         private WaveFormat _format;
+        private WaveFact _fact;
         private WaveData _data;
         private byte[] _buffer;
 
-        public int Size =>  (int)_data.SubChunkSize;
-        public int Fequency => (int)_format.SampleRate;
+        public int Size => _buffer.Length;
+        public int Frequency => (int)_format.SampleRate;
+        public int AudioFormat => _format.AudioFormat;
         public int Channels => _format.NumChannels;
         public int BitsPerSample => _format.BitsPerSample;
         public byte[] Buffer => _buffer;
 
-        public void Parse(BinaryReader reader)
+        private int ByteToInt16(byte[] packed)
         {
-            _header = RiffHeader.Parse(reader);
+            // This is always little endian, unlike the C# builtin for unpacking a byte array.
+            return (packed[1] << 8) | packed[0];
+        }
 
-            if (new string(_header.ChunkId) != "RIFF" ||
-                new string(_header.Format) != "WAVE")
+        private byte[] ParseMicrosoft(BinaryReader reader, int size)
+        {
+            return reader.ReadBytes(size);
+        }
+        
+        private byte[] ParseDviAdpcm(BinaryReader reader, int size, int samplesPerBlock)
+        {
+            var imaIndexTable = new int[16] {
```

Comment:
this can only be of size 8 since the table is repeating. When accessing the table just a bitwise or with 0x7 (which will cause that 0x8 becomes 0 etc. See [here](https://github.com/dbry/adpcm-xq/blob/master/adpcm-lib.c#L150) for a reference of this technique.